static/src/javascripts/projects/common/modules/component.spec.js

```diff
@@ -0,0 +1,418 @@
+// @flow
+
+import bean from 'bean';
+import fetchJSON_ from 'lib/fetch-json';
+
+import { Component } from './component';
+
+const fetchJSON: JestMockFn<*, *> = (fetchJSON_: any);
+
+const mockResponse = {
+    html: '<p>html</p>',
+    other: '<p>other</p>',
+};
+
+const createComponent = (props?: Object = {}): Component => {
+    const component = new Component();
+    const defaults = {
+        endpoint: 'whatever',
+    };
+
+    Object.assign(component, defaults, props);
+
+    return component;
+};
+
+jest.mock('lib/fetch-json', () => jest.fn(() => Promise.resolve(mockResponse)));
+
+jest.mock('bean', () => ({
+    off: jest.fn(),
+}));
+
+describe('Component', () => {
+    let elem;
+    let subElem;
+
+    beforeEach(() => {
+        if (document.body) {
+            document.body.innerHTML = `
+                <div class="component">
+                    <div class="component__element"></div>
+                </div>
+            `;
+        }
+
+        // Apologies for the typecast, but it makes things just way easier ...
```

Comment:
This is where `lib/$` becomes [highly valuable](https://github.com/guardian/frontend/issues/17467#issuecomment-344618114)